Update Python to 3.12 (latest)

heyLu opened the pull request:
Python 3.8 is a bit old at this point and a couple of dependencies already require Python 3.10, making the current image incompatible.

One examples is https://github.com/AVATEAM-IT-SYSTEMHAUS/mkdocs-kroki-plugin/blob/main/setup.py#L20, which my employer uses internally and thus we need a Python version update to be able to use the latest version of that plugin.

heyLu commented:
Trying again, @awanlin, can you review this or pass this along to someone who can?

awanlin reviewed (approved):
Thanks @heyLu, looks good to me 🚀

alexlorenzi commented:
Thanks for contribution @heyLu

heyLu commented:
@alexlorenzi will there be a release for this and other changes since the last release?  Upstream [seems to use](https://github.com/backstage/backstage/blob/85d9346ef11c1c20e4405102b4f5d93afb1292c1/plugins/techdocs-node/src/stages/generate/techdocs.ts#L56) the current release from last August, but an updated release would be nice I think.  (Or should we use the automatically built non-tagged images pushed by CI?)
